sdk/idp_oauth_access_token_source_test.go

```diff
@@ -0,0 +1,97 @@
+package sdk
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/opentdf/platform/lib/ocrypto"
+	"github.com/opentdf/platform/sdk/auth"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+)
+
+func TestNewOAuthAccessTokenSource_Success(t *testing.T) {
+	mockToken := "mockToken"
+	// Expected
+	mockSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: mockToken})
+	mockScopes := []string{"scope1", "scope2"}
+	mockKey, _ := ocrypto.NewRSAKeyPair(dpopKeySize)
+	dpopPublicKeyPEM, dpopKey, asymDecryption, _ := getNewDPoPKey(&mockKey)
+
+	// Testable
+	tokenSource, err := NewOAuthAccessTokenSource(mockSource, mockScopes, &mockKey)
+
+	// Sanity Checks
+	require.NoError(t, err)
+	assert.NotNil(t, tokenSource)
+	assert.Equal(t, mockSource, tokenSource.source)
+	assert.Equal(t, mockScopes, tokenSource.scopes)
+	// DPoP values
+	assert.Equal(t, asymDecryption, &tokenSource.asymDecryption)
+	assert.Equal(t, dpopPublicKeyPEM, tokenSource.dpopPEM)
+	assert.Equal(t, dpopKey, tokenSource.dpopKey)
+	// Interface checks
+	tok, err := tokenSource.AccessToken(context.Background(), nil)
+	require.NoError(t, err)
+	assert.Equal(t, tok, auth.AccessToken(mockToken))
+	made, err := tokenSource.MakeToken(func(jwk.Key) ([]byte, error) { return []byte(mockToken), nil })
+	require.NoError(t, err)
+	assert.Equal(t, made, []byte(mockToken))
+}
+
+func TestNewOAuthAccessTokenSource_ExpiredToken(t *testing.T) {
```

Comment:
It could be good to add a test for the empty string `""` token as well.  Right now it should be the same as the expired condition, but it will be a good case for the future when the TODO is implemented for refreshing the token.

Reply:
Thanks, and I'm not sure if we do want the SDK concerned with refreshing access tokens or if it's work an SDK consumer should do on their own.